Packs/EWS/Integrations/EWSO365/EWSO365.py

```diff
@@ -2143,35 +2142,51 @@ def fetch_emails_as_incidents(client: EWSClient, last_run):
     :return:
     """
     last_run = get_last_run(client, last_run)
-
+    excluded_ids = set(last_run.get(LAST_RUN_IDS, []))
     try:
         last_emails = fetch_last_emails(
             client,
             client.folder_name,
             last_run.get(LAST_RUN_TIME),
-            last_run.get(LAST_RUN_IDS),
+            excluded_ids,
         )
 
-        ids = deque(
-            last_run.get(LAST_RUN_IDS, []), maxlen=client.last_run_ids_queue_size
-        )
         incidents = []
         incident: Dict[str, str] = {}
+        demisto.debug(f'{APP_NAME} - Started fetch with {len(last_emails)} at {last_run.get(LAST_RUN_TIME)}')
+        current_fetch_ids = set()
         for item in last_emails:
             if item.message_id:
-                ids.append(item.message_id)
+                current_fetch_ids.add(item.message_id)
                 incident = parse_incident_from_item(item)
                 incidents.append(incident)
 
                 if len(incidents) >= client.max_fetch:
                     break
 
-        last_run_time = incident.get("occurred", last_run.get(LAST_RUN_TIME))
-        if isinstance(last_run_time, EWSDateTime):
-            last_run_time = last_run_time.ewsformat()
+        demisto.debug(f'{APP_NAME} - ending fetch - got {len(incidents)} incidents.')
+
+        last_fetch_time = last_run.get(LAST_RUN_TIME)
+
+        last_incident_run_time = incident.get("occurred", last_fetch_time)
```

Comment:
@JasBeilin / @esharf  are we certain that items in the loop in 2158 are ordered by time, so the last incident in the loop is really the latest mail that was iterated by the loop?

Reply:
yes
![image](https://user-images.githubusercontent.com/71636766/126793078-65c05d58-7a51-4457-a341-9cb8e758629f.png)

Reply:
```suggestion
        most_recent_incident_time = incident.get("occurred", last_fetch_time)
```